app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -65,38 +103,64 @@ class SearchActivity : AppCompatActivity() {
     }
 
     private fun clearButtonVisibility(s: CharSequence?): Int {
+        showMessage("",null, false)
         return if (s.isNullOrEmpty()) {
             View.GONE
         } else {
             View.VISIBLE
         }
     }
-}
 
+    private fun updateListTracks(){
+
+        if(!savedSearchEditText.isNullOrEmpty()){
+            playlistRetrofit.search(savedSearchEditText!!)
+                .enqueue(object: Callback<TrackResponse>{
+                    override fun onResponse(call: Call<TrackResponse>, response: Response<TrackResponse>) {
+                        if(response.isSuccessful){
+                            listTracks.clear()
+                            if(response.body()!!.results.isNotEmpty()){
```

Comment:
🍏 Здесь тоже опасно использовать !!, ответ от какого-нибудь сервиса может и с пустым body прийти. Лучше проверь на null, а затем используй.
```suggestion
                            if(response.body()?.results?.isNotEmpty() == true)
```